NixSupport/devenv.nix

```diff
@@ -0,0 +1,86 @@
+{ pkgs, config, lib, ... }:
+
+let
+  cfg = config.ihp;
+  types = lib.types;
+in
+{
+  options.ihp = {
+    enable = lib.mkEnableOption "Enable IHP support";
+
+    ghcCompiler = lib.mkOption {
+      default = pkgs.haskell.packages.ghc944;
+
+      description = ''
+        The GHC compiler to use for IHP.
+      '';
+    };
+
+    haskellPackages = lib.mkOption {
+      default = p: with p; [
+        cabal-install
+        base
+        wai
+        text
+        hlint
+        ihp
+      ];
+      description = ''
+        List of Haskell packages to be installed in the IHP environment.
+      '';
+    };
+
+    projectPath = lib.mkOption {
+      type = types.path;
+      description = ''
+        Path to the IHP project.
+      '';
+    };
```

Comment:
In order to seed the database with the application schema we must know its path to add these file to the nix store.

Reply:
This now triggers a copy of the full project directory to the nix store. We've had the same copy issue in the past, and walked around it by calling `toString` on the path: https://github.com/digitallyinduced/ihp/blob/master/NixSupport/default.nix#L14

The `toString` call turns it into an absolute path, avoding a full project copy.

Reply:
We actually have to copy the migration files to the nix-store in order to build the initial schema file for the postgres setup-script. 

But we can indeed avoid copying the whole project by building an absolute path to each file so that nix only add these to the store by using `${rootPath + "/relative/path.sql"}` instead of `${rootPath}/relative/path.sql`.